tests/kvm.rs

```diff
@@ -0,0 +1,201 @@
+mod config;
+
+use env_logger;
+use log::debug;
+use std::mem;
+use std::panic;
+use std::process::{Command, Stdio};
+use std::sync::{mpsc, Once};
+use std::thread;
+use std::time::Duration;
+
+use config::{KVMI_SOCKET, TIMEOUT, VIRSH_URI, VM_NAME};
+
+// to init env logger
+static INIT: Once = Once::new();
+
+fn run_test<T>(test: T) -> ()
+where
+    T: Send + 'static,
+    T: FnOnce() -> (),
+{
+    // init env_logger if necessary
+    INIT.call_once(|| {
+        env_logger::builder().is_test(true).init();
+    });
+    // setup before test
+    setup_test();
+
+    // setup test execution in a thread
+    let (done_tx, done_rx) = mpsc::channel();
+    let handle = thread::spawn(move || {
+        let val = test();
+        done_tx.send(()).expect("Unable to send completion signal");
+        val
+    });
+
+    // wait for test to complete until timeout
+    let timeout = Duration::from_secs(TIMEOUT);
```

Comment:
What about: https://docs.rs/ntest/0.3.3/ntest/attr.timeout.html

Reply:
There is also a possibility to set the timeout globally if you are simply worried about a blocking CI run. https://os.phil-opp.com/testing/#timeouts

Reply:
Thanks for the suggestion.
The issue i had with ntest is that the timeout is applied to the whole test, including setup and teardown.


However, if i can combine that with test-context, it can work

Reply:
Why is it a problem that setup and teardown are included in the timeout? Just account for that by increasing your timeout accordingly. 🤷‍♂️

Reply:
The setup time depends on the VM that's being tested.
It's not the same thing to revert a winxp 128MB snapshot and an 8GB RAM Windows 10 snapshot :)
And I didn't wanted to make the timeout depend on the environment, if possible

Reply:
Yeah, that makes sense.